src/main/java/com/exadel/discount/mapper/UserCityMapper.java

```diff
@@ -0,0 +1,24 @@
+package com.exadel.discount.mapper;
+
+import com.exadel.discount.dto.user.UserCityDto;
+import com.exadel.discount.entity.City;
+import com.exadel.discount.entity.Country;
+import com.exadel.discount.entity.User;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+
+@Mapper(componentModel = "spring")
+public interface UserCityMapper {
+
+    @Mapping(source = "user.id", target = "id")
+    @Mapping(source = "user.firstName", target = "firstName")
+    @Mapping(source = "user.lastName", target = "lastName")
+    @Mapping(source = "user.login", target = "login")
+    @Mapping(source = "user.password", target = "password")
+    @Mapping(source = "user.role", target = "role")
+    @Mapping(source = "user.phone", target = "phone")
+    @Mapping(source = "user.email", target = "email")
+    @Mapping(source = "city.name", target = "cityName")
+    @Mapping(source = "country.name", target = "country")
+    UserCityDto toUserCityDto(User user, City city, Country country);
+}
```

Comment:
Doesn't work

Reply:
Please, test just your branch with dev merged into it, it should work. Now there is a lot of compile errors

Reply:
Merged. Works...

Reply:
Done